Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapOfflineMapPreference.java

```diff
@@ -0,0 +1,361 @@
+        Resources res = context.getResources();
+        String text = String.format(res.getString(R.string.pref_baidu_map_offline_download_remove_message), cityid);
+        Toast.makeText(context, text, Toast.LENGTH_SHORT).show();
+        updateView();
+    }
+
+    /**
+     * Display offline map update status
+     */
+    public void updateView() {
+        mLocalMapList = mOffline.getAllUpdateInfo();
+        if (mLocalMapList == null) {
+            mLocalMapList = new ArrayList<MKOLUpdateElement>();
+        }
+        mLAdapter.notifyDataSetChanged();
+    }
+
+    public String formatDataSize(int size) {
+        String ret = "";
+        if (size < (DATA_SIZE_ONE_KB)) {
+            ret = String.format("%dK", size / DATA_SIZE_UNIT_KB);
+        } else {
+            ret = String.format("%.1fM", size / (DATA_SIZE_UNIT_MB));
+        }
+        return ret;
+    }
+
+    @Override
+    protected void onDialogClosed(boolean positiveResult) {
+        // destroy offline map module when exit
+        mOffline.destroy();
+        super.onDialogClosed(positiveResult);
+    }
+
+
+    @Override
+    public void onBindDialogView(View view){
+        // Initialize Baidu Map library
+        com.baidu.mapapi.SDKInitializer.initialize(this.getContext().getApplicationContext());
+        mBindDialogView = view;
+        mOffline = new MKOfflineMap();
+        mOffline.init(this);
+
+        initView(view);
+        super.onBindDialogView(view);
+    }
+
+    @Override
+    public void onGetOfflineMapState(int type, int state) {
+        switch (type) {
+            case MKOfflineMap.TYPE_DOWNLOAD_UPDATE: {
+                MKOLUpdateElement update = mOffline.getUpdateInfo(state);
+                // display download progress
+                if (update != null) {
+                    mStateView.setText(String.format("%s : %d%%", update.cityName, update.ratio));
+                    updateView();
+                }
+            }
+            break;
+            case MKOfflineMap.TYPE_NEW_OFFLINE:
+                // Prompt offline map update is available
+                Log.d("Baidu offline map", String.format("add offlinemap num:%d", state));
+                break;
+            case MKOfflineMap.TYPE_VER_UPDATE:
+                // Not used yet
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     *  list adapter for managing offline map
+     */
+    public class LocalMapAdapter extends BaseAdapter {
+        private View currentView = null;
+
+        @Override
+        public int getCount() {
+            return mLocalMapList.size();
+        }
+
+        @Override
+        public Object getItem(int index) {
+            return mLocalMapList.get(index);
+        }
+
+        @Override
+        public long getItemId(int index) {
+            return index;
+        }
+
+        @Override
+        public View getView(int index, View view, ViewGroup arg2) {
```

Comment:
@droneboost that's not what I meant. a `ListView` recycles views that are not visible so that it avoids creating new views as the user is scrolling up/down the list.
It does the recycling through the `View view` parameter in the `getView(int index, View view, ViewGroup args)` method. If there's no view to recycle, then that parameter is `null`, otherwise that parameter will be non null, and you can reuse it.

A typical example is as follow:
```
public View getView(int index, View view, ViewGroup args){
    if (view == null){
        view = View.inflate(...);
    }
    // else the 'view' parameter is non null, so let's reuse it directly
    initViewItem(view, ....);
    return view;
}
```

With the class field, you're overwriting the same view with the successive data, so all the entries in the listview will look identical.